pkg/issuer/acme/dns/digitalocean/digitalocean.go

```diff
@@ -59,6 +61,36 @@ func (c *DNSProvider) Present(domain, token, keyAuth string) error {
 		return err
 	}
 
+	skipCreation := false
+
+	// check if the record has already been created
+	domains, _, err := c.client.Domains.List(context.Background(), &godo.ListOptions{})
+	for _, domain := range domains {
+		if dns.Fqdn(domain.Name) == fqdn {
+			// loop over each record
+			for x := range dns.ParseZone(strings.NewReader(domain.ZoneFile), "", "") {
+				if x.Error != nil {
+					return x.Error
```

Comment:
Is `x.Error` a parse error for the zone file? Should we return an error here, or continue?

A few comments on each of the lines in this whole block would be very useful, plus maybe a paragraph explaining what we're doing here in english terms 😄

Reply:
If the zone file contains any parse errors, something's gone rather wrong. I feel the best thing to do in that case is exit and try again later. Thoughts?

Reply:
👍